.github/workflows/pr-realm-js.yml

```diff
@@ -440,90 +430,110 @@ jobs:
+          echo "wrapper=xvfb-run" >> $GITHUB_ENV
+
+      - name: Download TypeScript build
+        uses: actions/download-artifact@v4
         with:
-          path: '**/Pods'
-          key: ${{ runner.os }}-${{matrix.variant.environment}}-${{ hashFiles('**/Podfile.lock', './src/**', './vendor/**') }}
-          restore-keys: |
-            ${{ runner.os }}-${{matrix.variant.environment}}-
+          name: ts-build
 
-      - name: Setup Java
-        if: ${{ matrix.variant.os == 'android' }}
-        uses: actions/setup-java@v3
+      - name: Download prebuild native module
+        uses: actions/download-artifact@v4
         with:
-          distribution: 'zulu' # See 'Supported distributions' for available options
-          java-version: '${{ env.JAVA_VERSION }}'
+          name: ${{ matrix.prebuild }}
 
-      - name: Setup Android SDK
-        if: ${{ matrix.variant.os == 'android' }}
-        uses: android-actions/setup-android@v2
+      - name: Install dependencies
+        run: npm ci
+        env:
+          # Ensure we install the prebuild built in the previous job
+          npm_config_realm_local_prebuilds: ${{github.workspace}}/packages/realm/prebuilds
 
-      - name: Install NDK
-        if: ${{ matrix.variant.os == 'android' }}
-        run: sdkmanager --install "ndk;${{ env.NDK_VERSION }}"
+      - name: Start BaaS test server
+        id: baas
+        uses: ./.github/actions/baas-test-server
+        with:
+          branch: ${{ env.BAAS_BRANCH }}
+        env:
+          BAASAAS_KEY: ${{ secrets.BAASAAS_KEY }}
+
+      - name: Create Mocha Remote Context
+        id: mocha-env
+        run: echo "context=syncLogLevel=warn,longTimeoutMs=${{ env.LONG_TIMEOUT }},baseUrl=${{ steps.baas.outputs.baas-url }}" >> $GITHUB_OUTPUT
+
+      - name: Run tests
+        env:
+          MOCHA_REMOTE_CONTEXT: ${{ steps.mocha-env.outputs.context }}
+        # The non react native environments should not take so long
+        timeout-minutes: 60
+        run: ${{ env.wrapper }} npm run ${{ matrix.script_name}} --prefix integration-tests/environments/${{ matrix.environment }}
+
+  apple-tests:
+    name: Test iOS (React Native)
+    needs:
+      - generate-jsi
+      - build-ts
+      - prebuild-apple
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_14.3.1.app
+      IOS_DEVICE_NAME: iPhone 14
+    runs-on: macos-latest-large
+    timeout-minutes: 60
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+
+      - name: Setup node version
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+
+      # - name: Restore React Native cache
+      #   uses: actions/cache@v4
+      #   with:
+      #     path: '**/Pods'
+      #     key: ${{ runner.os }}-react-native-test-app-${{ hashFiles('**/Podfile.lock', './src/**', './vendor/**') }}
+      #     restore-keys: |
+      #       ${{ runner.os }}-react-native-test-app-
```

Comment:
We should re-enable this to avoid the extraction of boost while running `pod install`, as it takes ~ 4½ minutes 🤯

Reply:
I re-enabled this ...